### PR TITLE
adds `settings.aws.*` reference

### LINKED ISSUE #239: Document "AWS-specific settings"
**Path to page with content (_do not_ include domain name):**
`/en/os/1.14.x/api/settings/`

**What content is incorrect?**
The settings reference lacks the "[AWS-specific settings](https://github.com/bottlerocket-os/bottlerocket#aws-specific-settings)"

**What should the content say or communicate?**
There is an [ongoing effort](https://github.com/bottlerocket-os/bottlerocket-project-website/issues/203) to move the settings reference currently located on the [Bottlerocket README](https://github.com/bottlerocket-os/bottlerocket) to Bottlerocket.dev

This can be accomplished using the [Settings Documentation DSL](https://github.com/bottlerocket-os/bottlerocket-project-website/blob/main/METADOC-SETTING-DOCS-DSL.md). Effectively, the settings written out in the README's markdown needs to be converted to this DSL format. You can see an example of how this looks in the [ECS settings reference TOML file](https://github.com/bottlerocket-os/bottlerocket-project-website/blob/main/data/settings/1.14.x/ecs.toml) as well as [the actual ECS content markdown file](https://github.com/bottlerocket-os/bottlerocket-project-website/blob/main/content/en/os/1.14.x/api/settings/ecs/_index.markdown).

The file should be located at `/data/settings/1.14.x/aws.toml` and `/en/os/1.14.x/api/settings/aws/_index.markdown`.

### PR DESCRIPTION
**Pull request by stockholmux**
**Issue number:**

Closes #239

**Description of changes:**

Add setting reference for:
- `settings.aws.config`
- `settings.aws.profile`
- `settings.aws.credentials`
- `settings.aws.region`

**Terms of contribution:**

By submitting this pull request, I confirm that my contribution is made under
the terms of the licenses outlined in the LICENSE-SUMMARY file.

### CODE REVIEW

File: content/en/os/1.15.x/api/settings/aws/_index.markdown [hunk 1]
```diff
@@ -0,0 +1,11 @@
++++
+title="aws"
+type="docs"
+toc_hide=true
+description="Settings specific to the AWS platform (`settings.aws.*`)"
++++
+
+Only pertinent on `aws-*` variants, these settings are automatically gathered using instance metadata.
```
Comment (stmcginnis) [inline content/en/os/1.15.x/api/settings/aws/_index.markdown:8]: This is not actually accurate anymore. With [support added for IAM Roles Anywhere](https://github.com/bottlerocket-os/bottlerocket/pull/2553/files), even the VMware and Metal variants can use this in combination with the k8s ECR credential provider.

File: data/settings/1.15.x/aws.toml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+[[docs.ref.config]]
+description = """
+The base64-encoded representation of data used to populate `~/.aws/config`
+"""
+warning = """
+Avoid adding a `[profile default]` section.
+Recent versions of `aws-iam-authenticator` (and perhaps other components) not pick up the default credential settings when `settings.aws.profile` is set to `default`.
```
Comment (stmcginnis) [inline data/settings/1.15.x/aws.toml:7]:
```suggestion
Recent versions of `aws-iam-authenticator` (and perhaps other components) pick up the default credential settings when `settings.aws.profile` is set to `default`. 
```

File: content/en/os/1.15.x/api/settings/aws/_index.markdown (hunk 1 above)
Comment (stockholmux) [reply content/en/os/1.15.x/api/settings/aws/_index.markdown:8]: TIL. I'll revise!

Comment (stockholmux) [reply content/en/os/1.15.x/api/settings/aws/_index.markdown]: Fixed and adding cross-linking with with the setting in in `settings.kubernetes.credential-providers` (and fixed some formatting along the way).

**Review by stmcginnis — APPROVED** [review]

**Pull request merged**